geode-for-redis/src/main/java/org/apache/geode/redis/internal/data/RedisSet.java

```diff
@@ -110,6 +111,42 @@ private static MemberSet calculateDiff(RegionProvider regionProvider, List<Redis
     return diff;
   }
 
+  public static Set<byte[]> sinter(RegionProvider regionProvider, List<RedisKey> keys) {
+    List<RedisSet> sets = new ArrayList<>(keys.size());
+    RedisSet smallestSet = NULL_REDIS_SET;
+
+    for (RedisKey key : keys) {
+      RedisSet redisSet = regionProvider.getTypedRedisData(REDIS_SET, key, true);
+      if (redisSet == NULL_REDIS_SET) {
+        return Collections.emptySet();
+      }
+      if (smallestSet == NULL_REDIS_SET) {
+        smallestSet = redisSet;
+      }
+      if (smallestSet.scard() > redisSet.scard()) {
+        sets.add(smallestSet);
+        smallestSet = redisSet;
+      } else {
+        sets.add(redisSet);
+      }
+    }
+
+    MemberSet result = new MemberSet(smallestSet.scard());
+    for (byte[] member : smallestSet.members) {
+      boolean addToSet = true;
+      for (RedisSet otherSet : sets) {
+        if (!otherSet.members.contains(member)) {
+          addToSet = false;
+          break;
+        }
+      }
+      if (addToSet) {
+        result.add(member);
+      }
```

Comment:
not a big deal, but this can be simplified to
```
for(RedisSet otherSet : sets) {
    if(otherSet.members.contains(member)) {
        result.add(member);
    } else {
        break;
    }
}
```

Reply:
The intent of this loop is to ensure that the member is in **all** the sets (intersection). The suggested change would add the member if it appears in even one (of multiple) sets.